RFC-007-SWAP-Ethereum-basic.md

```diff
@@ -0,0 +1,315 @@
+    lt
+
+    // Jump to refund if time is expired
+    refund
+    jumpi
+
+exit:
+    // Exit
+    return(0, 0)
+
+redeem:
+    log1(0, 32, 0xB8CAC300E37F03AD332E581DEA21B2F0B84EAAADC184A295FEF71E81F44A7413) // log keccak256("Redeemed()")
+    selfdestruct(<redeem_identity>)
+
+refund:
+    log1(0, 0, 0x5D26862916391BF49478B2F5103B0720A842B45EF145A268F2CD1FB2AED55178) // log keccak256("Refunded()")
+    selfdestruct(<refund_identity>)
+}
+
+```
+
+The following is the contract deployment code encoded template as hex which will deploy the above contract when the placeholder values (set to `00`s) are replaced with the parameters.
+
+```
+6100dc61000f6000396100dc6000f336156051576020361415605c57602060006000376020602160206000600060026048f17f0000000000000000000000000000000000000000000000000000000000000000602151141660625760006000f35b42630000000010609f575b60006000f35b7fb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a741360206000a1730000000000000000000000000000000000000000ff5b7f5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed5517860006000a1730000000000000000000000000000000000000000
+```
+
+To compile the contract code replace the placeholder values with the HTLC paramters at the following offsets:
+
+
+| Data              | Position of first byte | Position of last byte | Length (bytes) |
+|:------------------|:-----------------------|:----------------------|:---------------|
+| `secret_hash`     | 51                     | 82                    | 32             |
+| `expiry`          | 99                     | 102                   | 4              |
+| `redeem_identity` | 153                    | 172                   | 20             |
+| `refund_identity` | 219                    | 248                   | 20             |
+
+The contract emits two logs:
+
+| Topic        | keccack256(Topic)                                                  | Data     | Emitted When             |
+|:-------------|:-------------------------------------------------------------------|----------|:-------------------------|
+| `Redeemed()` | 0xb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a7413 | `secret` | The contract is redeemed |
+| `Refunded()` | 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178 | -        | The contract is refunded |
+
+
+Implementations SHOULD use the following gas limits on the transactions related to the contract:
+
+| Transaction | recommended gas limit |
+|:------------|-----------------------|
+| Deployment  | 121,000               |
+| Redeem      | 100,000               |
+| Refund      | 100,000               |
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Ethereum blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract code as described in the previous section.
```

Comment:
```suggestion
At the start of the deployment stage, both parties construct the contract code as described in the previous section.
```
`compiles` implies conversion from EVM to hex. As we do not want people to do that, I think `construct` would be better (or anything that is not compile/build).

Reply:
I prefer compiles.